packages/mobile/src/send/RecipientPicker.tsx

```diff
@@ -111,24 +141,60 @@ export class RecipientPicker extends React.Component<RecipientProps> {
     </View>
   )
 
-  renderSendToPhoneNumber = (displayPhoneNumber: string, e164PhoneNumber: string) => {
-    const { t } = this.props
+  renderSendToPhoneNumber = (displayId: string, e164PhoneNumber: string) => {
+    const { t, onSelectRecipient } = this.props
     const recipient: RecipientWithMobileNumber = {
       kind: RecipientKind.MobileNumber,
       displayName: t('mobileNumber'),
-      displayPhoneNumber,
+      displayId,
       e164PhoneNumber,
     }
     return (
       <>
-        <RecipientItem recipient={recipient} onSelectRecipient={this.props.onSelectRecipient} />
+        <RecipientItem recipient={recipient} onSelectRecipient={onSelectRecipient} />
         {this.renderItemSeparator()}
       </>
     )
   }
 
+  renderSendToAddressOnly = () => {
+    const { t, onSelectRecipient, searchQuery } = this.props
+    const recipient: RecipientWithAddress = {
+      kind: RecipientKind.Address,
+      displayName: t('walletAddress'),
+      displayId: searchQuery.substring(2, 17) + '...',
+      address: searchQuery,
+    }
+
+    return (
+      <>
+        <RecipientItem recipient={recipient} onSelectRecipient={onSelectRecipient} />
+        {this.renderItemSeparator()}
+      </>
+    )
+  }
+
+  renderSendToAddress = () => {
```

Comment:
Nit: can we merge this with the function above? Or rename the function above to be more clear